File: 2014-10-CSharp-Fundamentals-1/6. Loops/IsPrime/IsPrime.cs
using System;

class IsPrime
{
    static void Main()
    {
        Console.Write("Enter a positive integer number: ");
        string numberAsString = Console.ReadLine();
        uint number = uint.Parse(numberAsString);
        uint divider = 2;
        uint maxDivider = (uint) Math.Sqrt(number);
        bool prime = true;
        while (prime && (divider <= maxDivider))
        {
            if (number % divider == 0)
            {
                prime = false;
            }
            divider++;
        }
        Console.WriteLine("Prime? {0}", prime);
    }
}

File: 2015-10-Java-Fundamentals-1/06. Loops/demos/IsPrime.java
import java.util.Scanner;

public class IsPrime {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
        
    System.out.print("Enter a positive integer number: ");
    int number = scanner.nextInt();
    int divider = 2;
    int maxDivider = (int)Math.sqrt(number);
    boolean prime = true;
        
    while (prime && (divider <= maxDivider)) {
      if (number % divider == 0) {
        prime = false;
      }
      
      divider++;
    }

    System.out.printf("Prime? %s%n", prime);
  }
}
